pandas/_testing/asserters.py

```diff
@@ -790,6 +790,16 @@ def assert_extension_array_equal(
         left_na, right_na, obj=f"{obj} NA mask", index_values=index_values
     )
 
+    # Specifically for StringArrayNumpySemantics, validate here we have a valid array
+    if isinstance(left.dtype, StringDtype) and left.dtype.storage == "python_numpy":
```

Comment:
Can we now just expose a property of the dtype that gives back the na_value and check that? I was thinking we could get rid of strings like "python_numpy" and "pyarrow_numpy"

Reply:
That property already exists (`StringDtype.na_value`), and should have used that here (the string "python_numpy" will indeed go away, have to update the constructor here after https://github.com/pandas-dev/pandas/pull/59330 is merged)